src/sbvr-api/permissions.ts

```diff
@@ -684,6 +685,201 @@ const getAlias = (name: string) => {
 	}
 	return `permissions${permissionsJSON}`;
 };
+
+const rewriteRelationship = memoizeWeak(
+	(
+		value: Relationship,
+		name: string,
+		abstractSqlModel: AbstractSqlModel,
+		permissionsLookup: PermissionLookup,
+		vocabulary: string,
+	) => {
+		let escapedName = sqlNameToODataName(name);
+		if (abstractSqlModel.tables[name]) {
+			escapedName = sqlNameToODataName(abstractSqlModel.tables[name].name);
+		}
+
+		const originalAbstractSQLModel = sbvrUtils.getAbstractSqlModel({
+			vocabulary,
+		});
+
+		const rewrite = (object: Relationship | RelationshipMapping) => {
+			if ('$' in object && Array.isArray(object.$)) {
+				// object is in the form of
+				// { "$": ["actor", ["actor", "id"]] } or { "$": ["device type"] }
```

Comment:
Can it ever be like `{ "$": ["device type"] }` ? The types say it cannot be but they could be wrong (and should be updated if so)

Reply:
Yes one example would be: 
```
{
 "device": {
  "has": {
    ...
    "uuid": { "$": ["uuid"] },
    ...
  }
  ...
 }
}
```

Reply:
Ok, lets update the types then to allow for this version

Reply:
done: https://github.com/balena-io-modules/abstract-sql-compiler/pull/83/files

Reply:
Approved